Tools/RunTopTagger.h

```diff
@@ -0,0 +1,235 @@
+            //std::cout << "  top type: " << type << std::endl;
+            
+            //std::cout << "  top type: " << type << std::endl;
+            if (   type == TopObject::Type::MERGED_TOP
+                || type == TopObject::Type::SEMIMERGEDWB_TOP
+                || type == TopObject::Type::RESOLVED_TOP)
+            {
+                AllTopsTLV->push_back(top->p());
+                if (type == TopObject::Type::MERGED_TOP)        MergedTopsTLV->push_back(top->p());
+                if (type == TopObject::Type::SEMIMERGEDWB_TOP)  SemiMergedTopsTLV->push_back(top->p());
+                if (type == TopObject::Type::RESOLVED_TOP)      ResolvedTopsTLV->push_back(top->p());
+                if (type == TopObject::Type::MERGED_W)          WTLV->push_back(top->p());
+
+                //print basic top properties (top->p() gives a TLorentzVector)
+                //N constituents refers to the number of jets included in the top
+                //3 for resolved tops 
+                //2 for W+jet tops
+                //1 for fully merged AK8 tops
+                
+                printf("\tTop properties: Type: %3d,   Pt: %6.1lf,   Eta: %7.3lf,   Phi: %7.3lf,   M: %7.3lf\n", static_cast<int>(top->getType()), top->p().Pt(), top->p().Eta(), top->p().Phi(), top->p().M());
+
+                //get vector of top constituents 
+                const std::vector<Constituent const *>& constituents = top->getConstituents();
+                std::vector<TLorentzVector> temp;
+
+                //Print properties of individual top constituent jets 
+                for(const Constituent* constituent : constituents)
+                {
+                    printf("\t\tConstituent properties: Constituent type: %3d,   Pt: %6.1lf,   Eta: %7.3lf,   Phi: %7.3lf\n", constituent->getType(), constituent->p().Pt(), constituent->p().Eta(), constituent->p().Phi());
+                    temp.push_back(constituent->p());
+                }                
+                TopJetsMap->insert(std::make_pair(topidx, temp));
+                ++topidx;
+            }
+        }
+
+        // number of tops
+        nAllTops        = AllTopsTLV->size();
+        nMergedTops     = MergedTopsTLV->size();
+        nSemiMergedTops = SemiMergedTopsTLV->size();
+        nResolvedTops   = ResolvedTopsTLV->size();
+        nWs             = WTLV->size();
+        
+        //print the number of tops found in the event 
+        //if (tops.size() > 1)
+        if (true)
+        {
+            printf("tops.size() =  %ld ",      tops.size());
+            printf("nAllTops =  %ld ",         nAllTops);
+            printf("nMergedTops =  %ld ",      nMergedTops);
+            printf("nSemiMergedTops =  %ld ",  nSemiMergedTops);
+            printf("nResolvedTops =  %ld ",    nResolvedTops);
+            printf("nWs =  %ld ",              nWs);
+            std::cout << std::endl;
+        }
+        
+        tr.registerDerivedVec("AllTopsTLV", AllTopsTLV);
```

Comment:
perhaps lets make these outputs match more closely what we have here?

https://github.com/susy2015/NanoSUSY-tools/blob/master/python/modules/DeepTopProducer.py#L208-L220